### PR TITLE
CDAP-15314 oracle db-plugin enhancement

### PR DESCRIPTION
**Pull request by ibarbashov**

### CODE REVIEW

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java [hunk 1]
```diff
@@ -46,9 +51,23 @@ public OracleSource(OracleSourceConfig oracleSourceConfig) {
...
   protected String createConnectionString(String host, Integer port, String database) {
+    if (OracleConstants.SERVICE_CONNECTION_TYPE.equals(this.connectionType)) {
+      return String.format(OracleConstants.ORACLE_CONNECTION_SERVICE_NAME_STRING_FORMAT, host, port, database);
+    }
+
     return String.format(OracleConstants.ORACLE_CONNECTION_STRING_FORMAT, host, port, database);
   }
 
+  @Path("getOracleSchema")
+  public Schema getOracleSchema(OracleSchemaRequest request,
```
Comment (albertshau) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java:62]: this way of getting schema is no longer supported and will fail to compile with 6.0.0. Whatever is set in configurePipeline is now what will be used. 

Don't have to fix in this PR, but let's be sure to fix it at some point in this repository. See https://github.com/cdapio/hydrator-plugins/blob/develop/database-plugins/src/main/java/io/cdap/plugin/db/batch/source/DBSource.java for how it is being done now.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java [hunk 2]
```diff
@@ -26,6 +26,9 @@ private OracleConstants() {
 
   public static final String PLUGIN_NAME = "Oracle";
   public static final String ORACLE_CONNECTION_STRING_FORMAT = "jdbc:oracle:thin:@%s:%s:%s";
+  public static final String ORACLE_CONNECTION_SERVICE_NAME_STRING_FORMAT = "jdbc:oracle:thin:@//%s:%s/%s";
   public static final String DEFAULT_BATCH_VALUE = "defaultBatchValue";
   public static final String DEFAULT_ROW_PREFETCH = "defaultRowPrefetch";
+  public static final String SERVICE_CONNECTION_TYPE = "service";
+  public static final String SID_CONNECTION_TYPE = "SID";
```
Comment (albertshau) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java]: is this used anywhere?

File: oracle-plugin/widgets/Oracle-batchsink.json [hunk 3]
```diff
@@ -39,9 +39,28 @@
             "default": "1521"
           }
         },
+        {
+          "name": "connectionType",
+          "label": "SID/Service Name",
```
Comment (albertshau) [inline oracle-plugin/widgets/Oracle-batchsink.json]: It looks weird if there are two properties with the same label. Do you know if there is some term that most people use for this? Do you think 'Connection Type' is hard to understand?

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java [hunk 4]
```diff
@@ -51,11 +51,19 @@ public OracleAction(OracleActionConfig oracleActionConfig) {
     @Nullable
     public Integer defaultBatchValue;
 
+    @Name("connectionType")
+    @Description("SID or Service Name")
```
Comment (albertshau) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: Whether to use an SID or Service Name when connecting to the database.

**Review by albertshau — COMMENTED** [review]
please update the docs too

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java (hunk 1 above)
Comment (ibarbashov) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java:62]: Ok

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java (hunk 2 above)
Comment (ibarbashov) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java]: removed

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java (hunk 4 above)
Comment (ibarbashov) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: added

File: oracle-plugin/widgets/Oracle-batchsink.json (hunk 3 above)
Comment (ibarbashov) [reply oracle-plugin/widgets/Oracle-batchsink.json]: Agree, "Connection Type" is clear to understand

**Review by albertshau — APPROVED** [review]
lgtm. Please squash changes and I'll merge.

**Review by albertshau — APPROVED** [review]

**Pull request merged**